hexa/workspaces/schema/mutations.py

```diff
@@ -107,55 +113,150 @@ def resolve_archive_workspace(_, info, **kwargs):
-        )
-        return {
-            "success": True,
-            "errors": [],
-            "workspace_membership": workspace_membership,
-        }
+                template_name="workspaces/mails/invite_member",
+                template_variables={
+                    "workspace": workspace.name,
+                    "owner": request.user.display_name,
+                    "workspace_url": request.build_absolute_uri(
+                        f"//{settings.NEW_FRONTEND_DOMAIN}/workspaces/{workspace.slug}"
+                    ),
+                },
+                recipient_list=[user.email],
+            )
+            return {
+                "success": True,
+                "errors": [],
+                "workspace_membership": workspace_membership,
+            }
+        except User.DoesNotExist:
+            invitation = WorkspaceInvitation.objects.create_if_has_perm(
+                principal=request.user,
+                workspace=workspace,
+                email=input["userEmail"],
+                role=input["role"],
+            )
+
+            token = invitation.generate_invitation_token()
+            send_mail(
+                title=gettext_lazy(
+                    f"You've been invited to join the workspace {workspace.name} on OpenHexa"
+                ),
+                template_name="workspaces/mails/invite_external_user",
+                template_variables={
+                    "workspace": workspace.name,
+                    "owner": request.user.display_name,
+                    "workspace_signup_url": request.build_absolute_uri(
+                        f"//{settings.NEW_FRONTEND_DOMAIN}/workspaces/{workspace.slug}/signup?email={input['userEmail']}&token={token}"
+                    ),
+                },
+                recipient_list=[input["userEmail"]],
+            )
+            return {
+                "success": True,
+                "errors": [],
+            }
+
     except Workspace.DoesNotExist:
         return {
             "success": False,
             "errors": ["WORKSPACE_NOT_FOUND"],
         }
-    except User.DoesNotExist:
+    except PermissionDenied:
         return {
             "success": False,
-            "errors": ["USER_NOT_FOUND"],
+            "errors": ["PERMISSION_DENIED"],
         }
-    except PermissionDenied:
+    except AlreadyExists:
         return {
             "success": False,
-            "errors": ["PERMISSION_DENIED"],
+            "errors": ["ALREADY_EXISTS"],
+        }
+
+
+@workspace_mutations.field("joinWorkspace")
+def resolver_join_workspace(_, info, **kwargs):
+    request: HttpRequest = info.context["request"]
+    input = kwargs["input"]
+
+    try:
+        signer = TimestampSigner()
+        decoded_value = b64decode(input["token"]).decode("utf-8")
+        # the token is valid for 48h
+        invitation_id = signer.unsign(decoded_value, max_age=48 * 3600)
+
+        invitation = WorkspaceInvitation.objects.get(id=invitation_id)
```

Comment:
I don't see a `catch WorkspaceInvitation.DoesNotExist`, I think it would be needed